### DISCUSSION TITLE
Is there public api to override some methods of AccessibleObject/Control

### DISCUSSION THREAD

**Discussion by beqabeqa473** [dotnet/winforms#10571] [Q&A]
Hello.

In a project i need to return an old behaviour of a listview object, because of lacking support of UIQA implementation of listview in NVDA screenreader in the past, and audience is from wider community, who may not upgrade to a current release NVDA version for specific reasons.

I am trying to do that, but unfortunately lots methods in AccessibleObject as well as Control is marked as internal like:
internal virtual bool IsIAccessibleExSupported
internal virtual bool SupportsUiaProviders => false;

Is there any api to override these methods, maybe proxy class?
Why it is not possible to customize a custom controll by a third-party developer?

Thanks.

**Comment by elachlan** [comment]
Hi @beqabeqa473, 

I believe the team are wanting to iron out all the issues with the implementation before making it a Public API.

@merriemcgaw might have more information. I am unsure if there is a tracking issue.

**Reply by merriemcgaw** [reply]
It's on our roadmap, but we don't have a specific timeline.

**Comment by beqabeqa473** [comment]
Frankly speaking, it is not good to have a possibility to do
customizations and don't have an access to it.

On 1/5/24, Lachlan Ennis ***@***.***> wrote:
> Hi @beqabeqa473,
>
> I believe the team are wanting to iron out all the issues with the
> implementation before making it a Public API.
>
> @merriemcgaw might have more information. I am unsure if there is a tracking
> issue.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/dotnet/winforms/discussions/10571#discussioncomment-8019549
> You are receiving this because you were mentioned.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Reply by elachlan** [reply]
Yes. But the API has to be stabilized before making it public. Otherwise you are stuck with the pattern or have to break it to fix it. 

If you would like to help speed up the finalization of the Accessibility APIs, you can help with issues lablled with `tenet-accessibility`:
https://github.com/dotnet/winforms/issues?q=is%3Aissue+is%3Aopen+label%3Atenet-accessibility

**Reply by elachlan** [reply]
I think the issue for what you want is being tracked here:
https://github.com/dotnet/winforms/issues/1834
